### PR TITLE
[MIG][15.0] coupon: migration for 15.0

### PR DESCRIPTION
**Pull request by daonamutc1**
This PR
Migration done

### CODE REVIEW

File: openupgrade_scripts/scripts/coupon/15.0.1.0/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+---Fields in module 'coupon'---
+coupon       / coupon.program           / company_id (many2one)         : now required, req_default: function
+# NOTHING TO DO: company_id has default
+
+---XML records in module 'coupon'---
+DEL ir.model.constraint: coupon.constraint_coupon_coupon_unique_coupon_code
+# NOTHING TO DO
+
+DEL mail.template: coupon.mail_template_sale_coupon [renamed to sale_coupon module] (noupdate)
+# DONE: post-migration: safely deleted xmlids
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/coupon/15.0.1.0/upgrade_analysis_work.txt:11]: wrong. you have to do xmlid_renames

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
I'm afraid this is not enough, as a total model renaming from `sale.coupon.*` to `coupon.*` has happened. I'm closing this and the `sale_coupon` ones to handle it.

**Comment by pedrobaeza** [comment]
Ouch, I have mixed v14 and v15. The renaming was done in v14. Anyway, this script is incorrect as pointed by Miquel, so I'm emitting the proper one.

**Pull request closed without merging**